### DISCUSSION TITLE
post_233649

### DISCUSSION THREAD

**Discussion by cincuranet** [cincuranet/blog-comments#1267] [Comments]
imported 

https://www.tabsoverspaces.com/id/233649

**Comment by cincuranet** [comment]
Imported comment written by _Lukáš Lánský_ on _2017-09-29T15:14:43_:

<p>0.0017 ns for Array.Count? Do you have a CPU that ticks in Thz? :-) ... Maybe something optimized the check away -- branch predictor, compiler, ...? It's tricky to measure such basic constructs.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-09-29T15:24:38_:

<p>I agree, it's tricky for these microbenchmarks. You can see that for array the standard deviation and the numbers jump all over the place. I took it as "fast as hell" for arrays. :) I forced the compiler to leave the code there. After that I wasn't investigating what the CPU was actually doing. Mostly to have it as a regular application code.</p>

**Comment by user1568891** [comment]
Thanks for the benchmarks! Could you run these again on the latest version of .Net?

**Comment by AleyCZ** [comment]
It seems there is a new rule in .NET 8 static code analysis requiring us to use Count instead of Any(). Microsoft is not making us happy, again. Because, of course, Any() shows intent and seems better. Why can't we use C# as a higher level language and must substitute the compiler in doing these inner optimizations? I feel like programming in assembler again when doing this...

**Reply by cincuranet** [reply]
You can easily disable that diagnostic, if you don't agree with it. No problem with that.
